e2e-tests/cypress/e2e/journeys/insurance/check-your-answers/your-business/change-your-answers/change-your-answers-change-company-details.spec.js

```diff
@@ -0,0 +1,295 @@
+  INSURANCE: {
+    EXPORTER_BUSINESS: {
+      COMPANY_HOUSE: {
+        INPUT,
+        COMPANY_NAME,
+        COMPANY_NUMBER,
+        COMPANY_INCORPORATED,
+        COMPANY_SIC,
+        FINANCIAL_YEAR_END_DATE,
+      },
+      YOUR_COMPANY: {
+        TRADING_ADDRESS,
+        TRADING_NAME,
+        WEBSITE,
+        PHONE_NUMBER,
+      },
+    },
+  },
+} = FIELD_IDS;
+
+const { taskList } = partials.insurancePartials;
+
+const task = taskList.submitApplication.tasks.checkAnswersAndSubmit;
+
+const { summaryList } = checkYourAnswersYourBusiness;
+
+const getFieldVariables = (fieldId, referenceNumber) => ({
+  route: COMPANY_DETAILS_CHECK_AND_CHANGE,
+  checkYourAnswersRoute: YOUR_BUSINESS,
+  newValueInput: '',
+  fieldId,
+  referenceNumber,
+  summaryList,
+  changeLink: summaryList.field(fieldId).changeLink,
+});
+
+context('Insurance - Check your answers - Company details - Your business - Summary list', () => {
+  let referenceNumber;
+  let url;
+
+  before(() => {
+    cy.completeSignInAndGoToApplication().then((refNumber) => {
+      referenceNumber = refNumber;
+
+      cy.completePrepareApplicationSinglePolicyType();
+
+      task.link().click();
+
+      // to get past eligibility check your answers page
+      submitButton().click();
+      // to get past policy and exports check your answers page
+      submitButton().click();
+
+      url = `${Cypress.config('baseUrl')}${ROOT}/${referenceNumber}${YOUR_BUSINESS}`;
+
+      cy.url().should('eq', url);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+
+    cy.navigateToUrl(url);
+  });
+
+  after(() => {
+    cy.deleteAccount();
+  });
+
+  describe(COMPANY_NUMBER, () => {
+    const fieldId = COMPANY_NUMBER;
+
+    let fieldVariables = getFieldVariables(fieldId, referenceNumber);
+
+    describe('when clicking the `change` link', () => {
+      beforeEach(() => {
+        cy.navigateToUrl(url);
+      });
+
+      it(`should redirect to ${COMPANY_DETAILS_CHECK_AND_CHANGE}`, () => {
+        cy.navigateToUrl(url);
+        fieldVariables = getFieldVariables(fieldId, referenceNumber);
+
+        summaryList.field(fieldId).changeLink().click();
+
+        cy.assertChangeAnswersPageUrl(referenceNumber, COMPANY_DETAILS_CHECK_AND_CHANGE, INPUT);
+      });
+    });
+
+    describe('form submission with a new answer', () => {
+      beforeEach(() => {
+        cy.navigateToUrl(url);
+
+        summaryList.field(fieldId).changeLink().click();
+
+        fieldVariables.newValueInput = '14440211';
+        changeAnswerField(fieldVariables, companyDetails.companiesHouseSearch());
+      });
+
+      it(`should redirect to ${YOUR_BUSINESS}`, () => {
+        cy.assertChangeAnswersPageUrl(referenceNumber, YOUR_BUSINESS);
+      });
+
+      it('should render the new answer', () => {
+        fieldVariables.newValue = fieldVariables.newValueInput;
+        checkChangeAnswerRendered(fieldVariables);
+
+        cy.checkText(summaryList.field(COMPANY_NAME).value(), 'AUDI LTD');
```

Comment:
Do we have to hard code these values? Is the company name the same as other tests where we need to tests 2 companies?